model/lifecycle.go

```diff
@@ -1355,6 +1345,42 @@ func getContainerFromRunOn(id string, buildVarTask BuildVariantTaskUnit, buildVa
 	return container, nil
 }
 
+// getContainerOptions resolves the task's container configuration based on the
+// task's container name and the container definitions available to the project.
+func getContainerOptions(project *Project, pRef *ProjectRef, container string) (*task.ContainerOptions, error) {
+	for _, c := range project.Containers {
+		if c.Name != container {
+			continue
+		}
+
+		opts := task.ContainerOptions{
```

Comment:
How about instead of doing this we change the model.Container struct to contain just Name and ContainerOpts as its two properties. Then we can just do `opts := c.ContainerOpts` and not need to duplicate similar looking structs. 

We'd still loop through the container sizes and overwrite the resources if we find a matching size.

Reply:
I prefer to avoid doing this for future-proofing reasons - using the same model for both would mixes YAML models and DB models together, so modifying the YAML structure will also affect the DB model (and vice-versa). Even if they're the same now, I usually write things to keep data models separate to avoid tight coupling.

I'm also assuming that that the container options for the DB model will become more complicated over time in ways that aren't related to YAML configuration.

Reply:
Gotcha. Sure if it'll give future flexibility SGTM